src/ort_genai_c.h

```diff
@@ -47,34 +60,154 @@ OGA_EXPORT size_t OGA_API_CALL OgaBufferGetDimCount(const OgaBuffer*);
 OGA_EXPORT OgaResult* OGA_API_CALL OgaBufferGetDims(const OgaBuffer*, size_t* dims, size_t dim_count);
 OGA_EXPORT const void* OGA_API_CALL OgaBufferGetData(const OgaBuffer*);
 
-OGA_EXPORT void OGA_API_CALL OgaDestroySequences(OgaSequences*);
-OGA_EXPORT size_t OGA_API_CALL OgaSequencesCount(const OgaSequences*);
-OGA_EXPORT size_t OGA_API_CALL OgaSequencesGetSequenceCount(const OgaSequences*, size_t sequence_index);
-OGA_EXPORT const int32_t* OGA_API_CALL OgaSequencesGetSequenceData(const OgaSequences*, size_t sequence_index);
+/*
+ * \param[in] sequences OgaSequences to be destroyed.
+ */
+OGA_EXPORT void OGA_API_CALL OgaDestroySequences(OgaSequences* sequences);
 
+/*
+ * \brief Returns the number of sequences in the OgaSequences
+ * \param[in] sequences
+ * \return The number of sequences in the OgaSequences
+ */
+OGA_EXPORT size_t OGA_API_CALL OgaSequencesCount(const OgaSequences* sequences);
+
+/*
+ * \brief Returns the number of tokens in the sequence at the given index
+ * \param[in] sequences
+ * \return The number of tokens in the sequence at the given index
+ */
+OGA_EXPORT size_t OGA_API_CALL OgaSequencesGetSequenceCount(const OgaSequences* sequences, size_t sequence_index);
+
+/*
+ * \brief Returns a pointer to the sequence data at the given index. The number of tokens in the sequence
+ *        is given by OgaSequencesGetSequenceCount
+ * \param[in] sequences
+ * \return The pointer to the sequence data at the given index. The sequence data is owned by the OgaSequences
+ *         and will be freed when the OgaSequences is destroyed. The caller must copy the data if it needs to
```

Comment:
Would it be better to explain based on lifetime, like:
"The pointer to the sequence data at the given index. The pointer is valid until the OgaSequences is destroyed. ..."